client/settings/payment-methods-list/payment-method.scss

```diff
@@ -44,10 +44,17 @@
 			margin: 0 4px 0 0;
 		}
 
+		.components-checkbox-control__label {
+			display: none;
+		}
```

Comment:
A sad workaround, because the version of Gutenberg we're bundling doesn't have support for the `hideLabelFromVision` prop